.github/workflows/validate_docker_images.yml

```diff
@@ -0,0 +1,37 @@
+name: Validate Docker Images (with Matrix Generation)
+
+on:
+  workflow_dispatch:
+    inputs:
+      channel:
+        description: 'PyTorch channel to use (nightly, test, release, all)'
+        required: true
+        type: choice
+        default: 'all'
+        options:
+          - 'nightly'
+          - 'test'
+          - 'release'
+          - 'all'
+  push:
+    branches:
+    - '*'
+
+jobs:
+  generate-matrix:
+    uses: pytorch/test-infra/.github/workflows/generate_docker_release_matrix.yml@main
+    with:
+      channel: ${{ inputs.channel }}
+    secrets: inherit
+
+  run-tests:
+    needs: generate-matrix
+    runs-on: ubuntu-latest
+    strategy:
+      matrix: ${{ fromJson(needs.generate-matrix.outputs.matrix) }}
+    container:
+      image: ghcr.io/pytorch/pytorch:2.2.2-cuda${{ matrix.cuda }}-cudnn${{ matrix.cudnn_version }}-${{ matrix.image_type }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run smoke tests
+        run: python test/smoke_test/smoke_test.py --package torchonly --runtime-error-check enabled
```

Comment:
Here before running tests we would need to set some env vars, like this so that smoke test actually runs some cuda tests:
```
run: |
export MATRIX_GPU_ARCH_TYPE="cuda"
export MATRIX_GPU_ARCH_VERSION="${{ matrix.cuda }}"
python test/smoke_test/smoke_test.py --package torchonly --runtime-error-check enabled
```